### PR TITLE
feat(moac): switch to grpc-js from grpc-uds

### PR DESCRIPTION
**Pull request by jkryl**
The workaround for UDS sockets in grpc lib is no longer needed so
we can get rid of grpc library that is obsolete and unmaintained.
It has been tested with nodejs v16 + k8s 1.20.7 with current versions
of csi-attacher and csi-provisioner.

It is also a big time saver when building moac image as we don't
have to compile the C++ grpc library in nix anymore.

grpc library in mayastor grpc tests stays and in addition to that
we use grpc-js as well. We will revisit that in the future and
convert everything to grpc-js. That has lower priority as it is
just a test code.

Resolves: CAS-927

### CODE REVIEW

**Review by gila — APPROVED** [review]

**Review by paulyoong — APPROVED** [review]

**Review by tiagolobocastro — APPROVED** [review]

File: chart/templates/moac-deployment.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ spec:
       serviceAccount: moac
       containers:
         - name: csi-provisioner
-          image: quay.io/k8scsi/csi-provisioner:v2.1.1
+          image: k8s.gcr.io/sig-storage/csi-provisioner:v2.2.1
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply chart/templates/moac-deployment.yaml:19]: Oh I see the mention now, it just was clear that it meant we're changing the image: "It has been tested with nodejs v16 + k8s 1.20.7 with current versions of csi-attacher and csi-provisioner."

[the remark this answers is not in the document]
Comment (jkryl) [reply chart/templates/moac-deployment.yaml:19]: good point. I will mention it in the commit message as that is easier to do at this stage 👍

File: csi/moac/node-composition.nix [hunk 2]
```diff
@@ -1,22 +1,20 @@
-# This file has been generated by node2nix 1.8.0. Do not edit!
+# This file has been generated by node2nix 1.9.0. Do not edit!
 
 { pkgs ? import <nixpkgs> {
     inherit system;
   }
 , system ? builtins.currentSystem
-, nodejs-slim ? pkgs.nodejs-slim-12_x
```
[the remark this answers is not in the document]
Comment (jkryl) [reply csi/moac/node-composition.nix:7]: I'm sorry, that was my oversight that I have fixed in the meantime and retested everything in k8s again. Should be ok now.

[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply csi/moac/node-composition.nix:7]: maybe we should keep a patch file in the tree that gets automatically applied or something like that, it's a bit of a pain to manually modify these files and very easy to forget

**Comment by jkryl** [comment]
bors try

**Comment by jkryl** [comment]
bors merge

**Pull request merged**